src/loader.js

```diff
@@ -284,6 +284,48 @@ function setupModuleLoader(window) {
 
           /**
            * @ngdoc method
+           * @name angular.Module#component
+           * @module ng
+           * @param {string} name Component name
+           * @param {Object} options Component definition object
+           * @description
+           * TODO...
+           */
+          component: function(name, options) {
+            function factory($injector) {
+              function makeInjectable(fn) {
+                if (angular.isFunction(fn)) {
+                  return function(tElement, tAttrs) {
+                    return $injector.invoke(fn, this, {$element: tElement, $attrs: tAttrs});
+                  };
```

Comment:
I wonder if this additional invoke via the `$injector` is an unwanted performance hit, since it would be called often, compared to the original factory function. But I guess that we can't get away from that given this API.

One alternative, could be to have a `dependencies` property on the `options`, which specifies all the dependencies that should be available in some closure to all these methods... Then we can inject these dependencies only once for all instances of the component, right?